code/edgebridge/src/main/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtension.java

```diff
@@ -178,7 +181,134 @@ private void dispatchTrackRequest(final Map<String, Object> data, final Event pa
+	 */
+	private Map<String, Object> formatData(Map<String, Object> data) {
+		// Create a mutable copy of data
+		Map<String, Object> mutableData = deepCopy(data);
+		// __adobe.analytics data container
+		Map<String, Object> analyticsData = new HashMap<>();
+
+		// Extract contextData
+		final Map<String, Object> contextData = DataReader.optTypedMap(
+			Object.class,
+			mutableData,
+			EdgeBridgeConstants.AnalyticsKeys.CONTEXT_DATA,
+			null
+		);
+		mutableData.remove(EdgeBridgeConstants.AnalyticsKeys.CONTEXT_DATA);
+		if (!isNullOrEmpty(contextData)) {
+			Map<String, Object> prefixedData = new HashMap<>();
+			Map<String, Object> nonPrefixedData = new HashMap<>();
+
+			for (Map.Entry<String, Object> entry : contextData.entrySet()) {
+				String key = entry.getKey();
+				Object value = entry.getValue();
+
+				// Check if the key starts with the specified prefix and add to corresponding map
+				if (key.startsWith(EdgeBridgeConstants.EdgeValues.PREFIX)) {
+					String newKey = key.substring(EdgeBridgeConstants.EdgeValues.PREFIX.length());
+					prefixedData.put(newKey, value);
+				} else {
+					nonPrefixedData.put(key, value);
+				}
+			}
+
+			// If there are prefixed data entries, add them to analyticsData
+			if (!prefixedData.isEmpty()) {
+				analyticsData.putAll(prefixedData);
+			}
+
+			// If there are non-prefixed data entries, add them under the contextData key
+			if (!nonPrefixedData.isEmpty()) {
+				analyticsData.put(EdgeBridgeConstants.EdgeKeys.CONTEXT_DATA, nonPrefixedData);
+			}
+		}
+
+		// Process action
+		Object actionValue = mutableData.remove(EdgeBridgeConstants.AnalyticsKeys.ACTION);
+		if (actionValue instanceof String) { // Validates non-null and correct type
+			String action = (String) actionValue;
+			analyticsData.put(EdgeBridgeConstants.EdgeKeys.LINK_NAME, action);
+			analyticsData.put(EdgeBridgeConstants.EdgeKeys.LINK_TYPE, EdgeBridgeConstants.EdgeValues.OTHER);
+		}
+
+		// Process state
+		Object stateValue = mutableData.remove(EdgeBridgeConstants.AnalyticsKeys.STATE);
+		if (stateValue instanceof String) { // Validates non-null and correct type
+			String state = (String) stateValue;
+			analyticsData.put(EdgeBridgeConstants.EdgeKeys.PAGE_NAME, state);
+		}
+
+		// If analyticsData is not empty, add it to mutableData under __adobe.analytics
+		if (!analyticsData.isEmpty()) {
+			Map<String, Object> adobeAnalytics = new HashMap<>();
+			adobeAnalytics.put(EdgeBridgeConstants.EdgeKeys.ANALYTICS, analyticsData);
+			mutableData.put(EdgeBridgeConstants.EdgeKeys.ADOBE, adobeAnalytics);
+		}
+
+		return mutableData;
+	}
+
+	/**
+	 * Creates a deep copy of the provided {@link Map}.
+	 *
+	 * @param map to be copied
+	 * @return {@link Map} containing a deep copy of all the elements in {@code map}
+	 */
+	Map<String, Object> deepCopy(final Map<String, Object> map) {
+		try {
+			return EventDataUtils.clone(map);
+		} catch (CloneFailedException e) {
+			Log.debug(
```

Comment:
shall we elevate this to warning? this should not happen unless something is incorrect